src/parquet/column/page.h

```diff
@@ -95,6 +95,21 @@ class DataPage : public Page {
   std::string min_;
 };
 
+class CompressedDataPage : public DataPage {
+ public:
+  CompressedDataPage(const std::shared_ptr<Buffer>& buffer, int32_t num_values,
```

Comment:
I had the exact line of thought. May be we can add a comment indicating that the codec information is stored in the parent `ColumnReader/ ColumnWriter`